x/epochstorage/README.md

```diff
@@ -0,0 +1,172 @@
+
+#### EpochDetails
+
+This object shows the details of the current epoch, the last epoch in memory and the last deleted epochs.
+
+```go
+type EpochDetails struct {
+	StartBlock    uint64   // start of the current epoch
+	EarliestStart uint64   // earliest epoch that is saved on memory
+	DeletedEpochs []uint64 // list of deleted epochs (updates each epoch)
+}
+```
+
+### FixatedParams
+
+Some params on the chain can affect the pairing list results, since pairing list needs to be the same throughout the epoch, the chain saves the needed params per epoch. this is done with fixated params. each module with a param that affects the pairing list needs to register the param as a fixated param using the following method of the keeper:
+
+```go
+AddFixationRegistry(fixationKey string, getParamFunction func(sdk.Context) any) 
+```
+
+by doing so the module will know to track [param change proposals](../spec/proposal_handler.go) and save any changes of the relevant param as a fixated param on the store.
+fixated param is in charge of tracking param changes, save their different versions if they are registered and delete old versions when they are not needed anymore (determined by EpochsToSave parameter).
+
+a fixation parameter is saved on store as:
+
+```go
+
+type FixatedParams struct {
+	Index         string // the index of the param
+	Parameter     []byte // serialized data of the param
+	FixationBlock uint64 // the block at which the param has changed
+}
+```
+
+this is done in the [begin block of the module](keeper/fixated_params.go)
+
+
+### StakeStorage
+
+StakeStorage is an object that is saved on the store per epoch. it contains the list of all the providers that are available in this epoch per chainID.
+
+```go
+type StakeStorage struct {
+	Index          string       // index of the stake storage (epoch + chainid)
+	StakeEntries   []StakeEntry // list of providers stake
+	EpochBlockHash []byte       // the block hash of the of the epoch block (used as salt for pairing)
+}
+```
+
+the index of the stakestorage is constructed from the epoch and chainID, for example: "100 ETH1"
+
+when a provider stakes himself to a chain (using the pairing module) he will be in the providers list of the next epoch. the stake storage of the next epoch (e.g. current stakestorage) is where all the changes are registered.
+
+when a new epoch starts, the module will create a copy of the current stake and save it as a specific epoch stake storage. this is done in the begin block of the module.
+
+also, when a new epoch starts the epoch storage will delete any outdated stakestorage (determined by the param EpochsToSave).
+note that if a stakestorage does not exist (either if it was deleted or it is in the future) pairing, verify pairing and relay payments will fail for the requested epoch (look at pairing readme).
```

Comment:
this sentence seems broken: "note that if a stakestorage does not exist (...) pairing" ??

Reply:
done

Reply:
instead of writing there "verify pairing" write "pairing verification"

Reply:
verify pairing is the correct term